src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -0,0 +1,445 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Scoring;
+using Microsoft.ML.OnnxScoring;
+using OnnxShape = System.Collections.Generic.List<long>;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+
+[assembly: LoadableClass(OnnxTransform.Summary, typeof(IDataTransform), typeof(OnnxTransform),
+    typeof(OnnxTransform.Arguments), typeof(SignatureDataTransform), OnnxTransform.UserName, OnnxTransform.ShortName)]
+
+[assembly: LoadableClass(OnnxTransform.Summary, typeof(IDataTransform), typeof(OnnxTransform),
+    null, typeof(SignatureLoadDataTransform), OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(OnnxTransform), null, typeof(SignatureLoadModel),
+    OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(OnnxTransform), null, typeof(SignatureLoadRowMapper),
+    OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: EntryPointModule(typeof(OnnxTransform))]
+
+namespace Microsoft.ML.OnnxScoring
+{
+    public sealed class OnnxTransform : ITransformer, ICanSaveModel
+    {
+        public sealed class Arguments : TransformInputBase
+        {
+
+            [Argument(ArgumentType.Required, HelpText = "Path to the onnx model file.", ShortName = "model", SortOrder = 0)]
+            public string ModelFile;
```

Comment:
>ModelFile [](start = 26, length = 9)

for ONNX, would you provide a single file or a directory ?

Reply:
For Onnx we only need a model_file. 

---
In reply to: [219906411](https://github.com/dotnet/machinelearning/pull/942#discussion_r219906411) [](ancestors = 219906411)